src/Paket.Core/Packaging/NupkgWriter.fs

```diff
@@ -83,23 +84,45 @@ module internal NupkgWriter =
             frameworkAssembliesList |> List.iter (buildFrameworkReferencesNode >> d.Add)
             metadataNode.Add d
 
-        let buildDependencyNode (Id, requirement:VersionRequirement) =
+        let buildDependencyNode (id, requirement:VersionRequirement) =
             let dep = XElement(ns + "dependency")
-            dep.SetAttributeValue(XName.Get "id", Id)
+            dep.SetAttributeValue(XName.Get "id", id)
             let version = requirement.FormatInNuGetSyntax()
             if String.IsNullOrEmpty version then
                 dep.SetAttributeValue(XName.Get "version", "0.0")
             else
                 dep.SetAttributeValue(XName.Get "version", version)
             dep
 
+        let buildGroupNode (framework:FrameworkIdentifier, add) = 
+            let g = XElement(ns + "group")
+            g.SetAttributeValue(XName.Get "targetFramework", framework.ToString())
+            add g
+            g
+
+
+        let buildDependencyNodes (excludedDependencies, add, dependencyList)  =
+            let takeFirstOfThree (a,_,_) = a
```

Comment:
That's a nice trick, but could you please manually inline this? I don't think we really need these functions here. Thx